core/src/main/java/org/infinispan/util/ByteString.java

```diff
@@ -0,0 +1,90 @@
+package org.infinispan.util;
+
+import org.infinispan.commons.marshall.AdvancedExternalizer;
+import org.infinispan.marshall.core.Ids;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Set;
+
+/**
+ * @author Tristan Tarrant
+ * @since 9.0
+ */
+public class ByteString {
```

Comment:
Why do you need this wrapper class? Can't you just work with byte[] directly? What's stopping you?